### PR TITLE
Mfa guides

### PR DESCRIPTION
**Pull request by sonalkr132**

### CODE REVIEW

File: stylesheets/application.css [hunk 1]
```diff
@@ -910,4 +910,11 @@ a.page__heading {
     .nav--v__link--footer:last-child {
       margin-bottom: 36px; } }
 
+.t-body .t-img {
```
Comment (sonalkr132) [inline stylesheets/application.css]: will add this as scss in seperate PR. also, need to fix this https://github.com/rubygems/guides/commit/c1f6b2ef710bb1b203aa100cdafd52244fd2d49b#diff-33530e60b01f198d2a5b4e0160101137

File: _layouts/default.html [hunk 2]
```diff
@@ -89,6 +89,8 @@
               <a class="nav--v__link" href="/rubygems-org-api">RubyGems.org API</a>
               <a class="nav--v__link" href="/rubygems-org-api-v2">RubyGems.org API V2.0</a>
               <a class="nav--v__link" href="/run-your-own-gem-server">Run your own gem server</a>
+              <a class="nav--v__link" href="/setting-up-multifactor-authentication" >Setting up multifactor authentication</a>
```
Comment (olleolleolle) [inline _layouts/default.html]:
```suggestion
              <a class="nav--v__link" href="/setting-up-multifactor-authentication">Setting up multifactor authentication</a>
```

File: _layouts/default.html [hunk 3]
```diff
@@ -89,6 +89,8 @@
...
+              <a class="nav--v__link" href="/using-mfa-in-command-line" >Using MFA in command line</a>
```
Comment (olleolleolle) [inline _layouts/default.html]:
```suggestion
              <a class="nav--v__link" href="/using-mfa-in-command-line">Using MFA in command line</a>
```

File: setting-up-multifactor-authentication.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+url: /setting-up-multifactor-authentication
+previous: /run-your-own-gem-server
+next: /using-mfa-in-command-line
+---
+
+<em class="t-gray">Want to better protect your rubygems.org account?</em>
+
+Your rubygems.org account is important! Unauthorized access of your account
+can lead to irrevocable damage to your gem's reputation. We highly recommend
+that you enable MFA for both UI and API. When enabled, we will ask you to provide an
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
that you enable MFA for both UI and API. When enabled, we will ask you to provide a
```

File: setting-up-multifactor-authentication.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+one time password for operations like login, gem push etc to verify your
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
one-time password for operations like login, gem push etc to verify your
```

File: setting-up-multifactor-authentication.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+identity.
+
+## Prerequisite
+
+You should have an authenticator app (like [Google Authenticator](https://support.google.com/accounts/answer/1066447) or [Authy](https://authy.com/download/)) which
+supports time-based one-time password (TOTP) to scan the QR-code and generate
+access code. SMS-based authentication or recovery is **not** supported.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
an access code. SMS-based authentication or recovery is **not** supported.
```

File: setting-up-multifactor-authentication.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+## Enabling multi-factor authentication
+
+1. Login to rubygems.org using your existing account and go to [edit profile](https://rubygems.org/profile/edit) page.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
1. Login to rubygems.org using your existing account and go to the [edit profile](https://rubygems.org/profile/edit) page.
```

File: setting-up-multifactor-authentication.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+Click _register a new device_ under _multifactor authentication_ section.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
Click _register a new device_ in the _multifactor authentication_ section.
```

File: setting-up-multifactor-authentication.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    ![Multifactor authentication section at profile edit page](/images/enabling_mfa_step1.png){:class="t-img"}
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
    ![Multifactor authentication section on the profile edit page](/images/enabling_mfa_step1.png){:class="t-img"}
```

File: setting-up-multifactor-authentication.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+2. You will be redirected to a page with QR-code and a text box for verifying OTP
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
2. You will be redirected to a page with a QR code and a text box for verifying OTP
```

File: setting-up-multifactor-authentication.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+code. Please use your authenticator to scan the QR-code. A new account for rubygems.org will be
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
code. Please use your authenticator to scan the QR code. A new account for rubygems.org will be
```

File: setting-up-multifactor-authentication.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+added to your authenticator app as soon as the scan completes.
+You can also add a new account manually using _Account_ and _Key_ shown next to QR code.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
You can also add a new account manually using _Account_ and _Key_ shown next to the QR code.
```

File: setting-up-multifactor-authentication.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+Please make sure you choose _time based_ option as MFA type.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
Please make sure you choose the option _time based_ as MFA type.
```

File: setting-up-multifactor-authentication.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+On successful registration, you will see a 6-digit access code (30
+seconds expiry) in your authenticator app for your rubygems.org account.
+Enter the shown access code in _OTP Code_ text field and click **enable**.
+    ![Registering a new device](/images/enabling_mfa_step2.png){:class="t-img"}
+3. If the code is correct and QR-code has not expired, on next page you will see a list of recovery
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
3. If the code is correct and the QR code has not expired, on next page you will see a list of recovery
```

File: setting-up-multifactor-authentication.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+codes. Please copy and store these codes in a safe place. You can use these recovery
+code to access your account, should you ever lose your phone or accidentally deleted
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
code to access your account, should you ever lose your phone or accidentally delete the
```

File: setting-up-multifactor-authentication.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+rubygems.org account from your authenticator app. Note that, each recovery code can be used
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
rubygems.org account from your authenticator app. Note that each recovery code can be used
```

File: setting-up-multifactor-authentication.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+only once. Please reregister your authenticator app after using recovery code to
+login to rubygems.org.
+    ![Recovery codes](/images/enabling_mfa_step3.png){:class="t-img"}
+4. Sign out and sign in again. Sign in page will now ask for OTP code.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
4. Sign out and sign in again. The sign-in page will now ask for an OTP code.
```

File: setting-up-multifactor-authentication.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    ![OTP prompt at login page](/images/mfa_login.png){:class="t-img"}
+
+## Authentication levels
+
+When you register a new device or enable MFA for the first time, we will enable
+MFA only for UI. If you go to _profile edit_ page again, in _multifactor authentication_ section, you
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
MFA only for UI. If you go to _profile edit_ page again, to the _multifactor authentication_ section, you
```

File: setting-up-multifactor-authentication.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+will see a dropdown menu with three options:
+
+- **Disabled**: disables MFA. Please delete rubygems.org account from you authenticator app after disabling.
+- **UI only**: sign in from browser and disabling MFA will require OTP code.
+- **UI and API**: gem signin, push and owner --add/remove will require OTP code.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
- **UI and API**: gem signin, push and owner --add and owner --remove will require OTP code.
```

File: setting-up-multifactor-authentication.md [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+
+Steps to change your MFA level:
+
+1. Sign in and go to _edit profile_ page. If you have enabled MFA for your account,
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
1. Sign in and go to the _edit profile_ page. If you have enabled MFA for your account,
```

File: setting-up-multifactor-authentication.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+under _multifactor authentication_ section, you will see a dropdown menu.
```
Comment (olleolleolle) [inline setting-up-multifactor-authentication.md]:
```suggestion
in the _multifactor authentication_ section, you will see a dropdown menu.
```

File: using-mfa-in-command-line.md [hunk 22]
```diff
@@ -0,0 +1,46 @@
+---
+layout: default
+title: Using multifactor authentication in command line
+url: /using-mfa-in-command-line
+previous: /setting-up-multifactor-authentication
+next: /resources
+---
+
+When your MFA level is _UI and API_, we will ask to you provide an OTP
+for `gem signin`, `gem push`, `gem owner --add/remove`. Check [setting up multifactor authentication](/setting-up-multifactor-authentication)
```
Comment (olleolleolle) [inline using-mfa-in-command-line.md]:
```suggestion
for `gem signin`, `gem push`, `gem owner --add`, `gem owner --remove`. Check [setting up multifactor authentication](/setting-up-multifactor-authentication)
```

File: using-mfa-in-command-line.md [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+for enabling MFA.
+
+You can preemptively pass otp code using `--otp` flag or else we will prompt
```
Comment (olleolleolle) [inline using-mfa-in-command-line.md]:
```suggestion
You can preemptively pass an OTP code using `--otp` flag or else we will prompt
```

File: using-mfa-in-command-line.md [hunk 24]
```diff
@@ -0,0 +1,46 @@
...
+    $ gem signin --otp 111111
+    Enter your RubyGems.org credentials.
+    Don't have an account yet? Create one at https://rubygems.org/sign_up
+    Email:   gem_author@example
+    Password:
+
+    Signed in.
+
+Note that `gem signin` only fetches and stores your rubygems.org api key. `gem signin`
+is not equivalent of creating a user session. We will check for OTP code everytime you
```
Comment (olleolleolle) [inline using-mfa-in-command-line.md]:
```suggestion
is not equivalent to creating a user session. We will check for OTP code every time you
```

File: using-mfa-in-command-line.md [hunk 25]
```diff
@@ -0,0 +1,46 @@
...
+use any of commands mentioned above.
```
Comment (olleolleolle) [inline using-mfa-in-command-line.md]:
```suggestion
use any of the commands mentioned above.
```

File: using-mfa-in-command-line.md [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+
+Publishing a gem after sign in from cli:
```
Comment (olleolleolle) [inline using-mfa-in-command-line.md]:
```suggestion
Publishing a gem after signing in from cli:
```

**Pull request merged**